src/expressions.md

```diff
@@ -41,28 +41,33 @@
 > &nbsp;&nbsp; &nbsp;&nbsp; | [_MatchExpression_]\
 > &nbsp;&nbsp; )
 
-An expression may have two roles: it always produces a *value*, and it may have
-*effects* (otherwise known as "side effects"). An expression *evaluates to* a
-value, and has effects during *evaluation*. Many expressions contain
-sub-expressions, called the *operands* of the expression. The meaning of each
-kind of expression dictates several things:
+Expressions are the fundamental unit of computation.
+When evaluated, they may have *effects* and evaluates to either a value or place.
+
+Where expressions are allowed are *expression contexts*.
+Expressions commonly contain sub-expressions, called the *operands* of the expression.
+
+The meaning of each kind of expression dictates several things:
 
 * Whether or not to evaluate the operands when evaluating the expression
 * The order in which to evaluate the operands
-* How to combine the operands' values to obtain the value of the expression
+* How to combine the operands' values to obtain the value or place of the expression
 
 In this way, the structure of expressions dictates the structure of execution.
-Blocks are just another kind of expression, so blocks, statements, expressions,
-and blocks again can recursively nest inside each other to an arbitrary depth.
+Blocks are just another kind of expression, so blocks, statements, expressions, and blocks again can recursively nest inside each other to an arbitrary depth.
 
-> **Note**: We give names to the operands of expressions so that we may discuss
-> them, but these names are not stable and may be changed.
+> **Note**: We give names to the operands of expressions so that we may discuss them, but these names are not stable and may be changed.
 
 ## Expression precedence
 
-The precedence of Rust operators and expressions is ordered as follows, going
-from strong to weak. Binary Operators at the same precedence level are grouped
-in the order given by their associativity.
+The grammar for expressions as presented in this book is wrong.
+It purposefully ignores that certain operands cannot be certain expressions due to precedence.
+For example, `a * b + c` is actually parsed as an addition operator expression with the multiplication operator expression as its left operand.
+The grammar, taking precedence into account, is too unweildy to read and understand.
```

Comment:
```suggestion
The grammar, taking precedence into account, is too unwieldy to easily read and understand.
```

Spell fix and insert `easily` since the grammar *can* be understood, it just takes effort.